osfoffline/application/main.py

```diff
@@ -127,55 +131,36 @@ def start(self):
             os.makedirs(user.osf_local_folder_path)
 
         self.start_tray_signal.emit()
-        logging.debug('starting background worker from main.start')
-        # fix the multithreading problem
-        try:
-            self.background_worker = BackgroundWorker()
-        except AttributeError:
-            pass
+        logger.debug('starting background worker from main.start')
+
+        self.background_worker = BackgroundWorker()
         self.background_worker.start()
 
     def resume(self):
-        logging.debug('resuming')
-        # todo: properly pause the background thread
-        # I am recreating the background thread everytime for now.
-        # I was unable to correctly pause the background thread
-        # thus took this route for now.
-        if self._can_restart_background_worker():
-            # stop previous
-            self.background_worker.stop()
-            self.background_worker.join()
+        logger.debug('resuming')
+        if self.background_worker.is_alive():
+            raise RuntimeError('Resume called without first calling pause')
 
-            # start new
-            self.background_worker = BackgroundWorker()
-            self.background_worker.start()
-        else:
-            # todo: what goes here, if anything?
-            logging.info('wanted to but could not resume background worker')
+        self.background_worker = BackgroundWorker()
+        self.background_worker.start()
 
     def pause(self):
-        logging.debug('pausing')
+        logger.debug('pausing')
         if self.background_worker and self.background_worker.is_alive():
-            logging.debug('pausing background worker')
             self.background_worker.stop()
-            logging.debug('background worker stopped')
-            self.background_worker.join()
-            logging.debug('paused background worker')
 
     def quit(self):
         try:
-            self.background_worker.pause_background_tasks()
-            self.background_worker.stop()
+            if self.background_worker.is_alive():
+                logger.info('Stopping background worker')
+                self.background_worker.stop()
 
+            logger.info('Saving user data')
             user = session.query(User).filter(User.logged_in).one()
             save(session, user)
             session.close()
-
-            QApplication.instance().quit()
-        except Exception as e:
-            # FIXME: Address this issue
-            logging.warning('quit broke. stopping anyways. Exception was {}'.format(e))
-            # quit() stops gui and then quits application
+        finally:
```

Comment:
This is a `try/finally` that doesn't appear to handle the exception; it's not clear to me that line 164 would be reached, in which case the log message may be a tad optimistic. Is something in this catchall fails, wouldn't the application still appear to hang and require a force-quit?